geodiff/src/drivers/sqlitedriver.cpp

```diff
@@ -49,20 +49,29 @@ class Sqlite3DbMutexLocker
 class Sqlite3SavepointTransaction
 {
   public:
-    explicit Sqlite3SavepointTransaction( std::shared_ptr<Sqlite3Db> db )
-      : mDb( db )
+    explicit Sqlite3SavepointTransaction( std::shared_ptr<Sqlite3Db> db, const Context *context )
+      : mDb( db ), mContext( context )
     {
       if ( sqlite3_exec( mDb.get()->get(), "SAVEPOINT changeset_apply", 0, 0, 0 ) != SQLITE_OK )
-        throw GeoDiffException( "Unable to start savepoint" );
+      {
+        std::string errMsg = sqliteErrorMessage( mDb.get()->get(), "Sqlite3SavepointTransaction" );
+        throw GeoDiffException( "Unable to start savepoint: " + errMsg );
+      }
     }
 
     ~Sqlite3SavepointTransaction()
     {
       if ( mDb )
       {
         // we had some problems - roll back any pending changes
-        sqlite3_exec( mDb.get()->get(), "ROLLBACK TO changeset_apply", 0, 0, 0 );
-        sqlite3_exec( mDb.get()->get(), "RELEASE changeset_apply", 0, 0, 0 );
+        if ( sqlite3_exec( mDb.get()->get(), "ROLLBACK TO changeset_apply", 0, 0, 0 ) != SQLITE_OK )
+        {
+          logSqliteError( mDb, mContext->logger(), "~Sqlite3SavepointTransaction" );
```

Comment:
in the logSqlite3Error 3rd argument, could we add more context to all calls?? so if you see a debug log in half year you can clearly uindicarte where it come from/on which operation